core/app/models/spree/store_credit.rb

```diff
@@ -288,4 +289,11 @@ def associate_credit_type
       self.credit_type = Spree::StoreCreditType.find_by_name(credit_type_name)
     end
   end
+
+  def apply_to_orders_in_confirm_state
```

Comment:
I'm fine with restart_checkout_flow, but not with checking for the "confirm" state. Sorry if that wasn't clear initially.